### ISSUE TITLE
Magento 2.3.4 couldn't get category image name

### ISSUE THREAD

**Issue by Aravindh468** [magento/magento2#27168]
### Preconditions (*)

1. Magento 2.3.4

### Steps to reproduce (*)

1. Create a Category and add an image to that category.
2. I need to display the category image of few categories based on id in homepage. So i have try to use the code in frontend as below,

```
$category->getImageUrl()

$category->getImage()
```

3. But it returns the tmp path image of that category.
4. I have even tried using the above codes for getting category image name in **vendor/magento/module-catalog/view/frontend/templates/category/image.phtml** but end up with getting wrong result.

### Expected result (*)

1. $category->getImageUrl() - It should return full path of category image(localhost/ver1/pub/media/catalog/category/imagename.jpg)

2. $category->getImage()- It should return only the name of category image(imagename.jpg)

### Actual result (*)

1. Both the code returns temporary(tmp) path image of the category (/localhost/ver1/pub/media/catalog/tmp/category/imagename.jpg)

 2.Couldn't get the filename of the category image separately.

**Comment by Aravindh468** [comment]
I have found the issue is because category backend model image before save method 

> vendor/magento/module-catalog/Model/Category/Attribute/Backend/Image.php

```
/**
     * Avoiding saving potential upload data to DB.
     *
     * Will set empty image attribute value if image was not uploaded.
     *
     * @param \Magento\Framework\DataObject $object
     * @return $this
     * @since 101.0.8
     */
    public function beforeSave($object)
    {
        $attributeName = $this->getAttribute()->getName();
        $value = $object->getData($attributeName);

        if ($this->fileResidesOutsideCategoryDir($value)) {
            // use relative path for image attribute so we know it's outside of category dir when we fetch it
            // phpcs:ignore Magento2.Functions.DiscouragedFunction
            $value[0]['url'] = parse_url($value[0]['url'], PHP_URL_PATH);
            $value[0]['name'] = $value[0]['url'];
        }

        if ($imageName = $this->getUploadedImageName($value)) {
            if (!$this->fileResidesOutsideCategoryDir($value)) {
                $imageName = $this->checkUniqueImageName($imageName);
            }
            $object->setData($this->additionalData . $attributeName, $value);
            $object->setData($attributeName, $imageName);
        } elseif (!is_string($value)) {
            $object->setData($attributeName, null);
        }

        return parent::beforeSave($object);
    }
```
```

/**
     * Check for file path resides outside of category media dir. The URL will be a path including pub/media if true
     *
     * @param array|null $value
     * @return bool
     */
    private function fileResidesOutsideCategoryDir($value)
    {
        if (!is_array($value) || !isset($value[0]['url'])) {
            return false;
        }

        $fileUrl = ltrim($value[0]['url'], '/');
        $baseMediaDir = $this->_filesystem->getUri(DirectoryList::MEDIA);

        if (!$baseMediaDir) {
            return false;
        }

        return strpos($fileUrl, $baseMediaDir) !== false;
    }
```
In the above method in condition that has been returned checked as below,

> strpos($fileUrl, $baseMediaDir) !== false;

It should be,

> strpos($fileUrl, $baseMediaDir) === false;

I have checked in previous versions.

**Comment by engcom-Charlie** [comment]
Hello @Aravindh468 @sdzhepa
This issue looks like a feature request.
@sdzhepa could you transfer this issue to the Feature requests repository?
Thanks for your collaboration!

**Comment by ihor-sviziev** [comment]
Hi @engcom-Charlie,
This is not a feature, but a bug. Production image shouldn't be stored in temporary (tmp) folder, they have to be moved.

**Comment by engcom-Charlie** [comment]
Hi @Aravindh468, @ihor-sviziev
I've checked this issue on the latest 2.4-develop and I get image path like `/pub/media/catalog/category/{{image}}` as expected.
And what about `$category->getImage()` should return image name without path. It wasn't implemented before so it looks like a feature request, does it?

**Comment by ihor-sviziev** [comment]
@engcom-Charlie could you check it on 2.3.4, to be sure it was fixed in 2.4-develop?

**Comment by ihor-sviziev** [comment]
@magento give me 2.3.4 instance

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev. Thank you for your request. I'm working on Magento 2.3.4 instance for you

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, here is your [Magento instance](https://i-27168-2-3-4.instances.magento-community.engineering).
Admin access: [https://i-27168-2-3-4.instances.magento-community.engineering/admin_0c2d](https://i-27168-2-3-4.instances.magento-community.engineering/admin_0c2d)
Login: `114e8128` Password: `2529273996a5`
Instance will be terminated in up to 3 hours.

**Comment by ihor-sviziev** [comment]
**Tested on 2.3.4:**
❌Category Image contains "tmp" in the URL
https://i-27168-2-3-4.instances.magento-community.engineering/category-1.html
![image](https://user-images.githubusercontent.com/1873745/83876144-0661b180-a741-11ea-8dfc-4d987ddfe10c.png)

Image URL is following:
https://i-27168-2-3-4.instances.magento-community.engineering/pub/media/catalog/tmp/category/Tests.png

**Comment by ihor-sviziev** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, here is your [Magento instance](https://i-27168-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-27168-2-4-develop.instances.magento-community.engineering/admin_c9bf](https://i-27168-2-4-develop.instances.magento-community.engineering/admin_c9bf)
Login: `e37f3a73` Password: `6e0322912876`
Instance will be terminated in up to 3 hours.

**Comment by ihor-sviziev** [comment]
**Tested on 2.4-develop:**
✔Category Image do not contains "tmp" in the URL
https://i-27168-2-4-develop.instances.magento-community.engineering/category-1.html
![image](https://user-images.githubusercontent.com/1873745/83876618-d23ac080-a741-11ea-8ac5-7abe1233ba9b.png)

Image URL is following:
https://i-27168-2-4-develop.instances.magento-community.engineering/pub/media/catalog/category/Tests.png


This issue was fixed in 2.4-develop branch and not reproducing anymore.
I'm closing this issue.
